tests/check_check_tags.c

```diff
@@ -0,0 +1,621 @@
+	      ret_val, strlen(expected_str));
+    
+    ck_assert_msg(strcmp(ret_str, expected_str) ==  0,
+	      "Got '%s' instead of '%s'",
+	      ret_str, expected_str);
+} END_TEST;
+
+START_TEST(set_get_truncated)
+{
+    char ret_str[10];
+    const char *actual_str = "Red Blue Green Yellow";
+    char *expected_str;
+    int ret_val;
+    TCase *tc;
+
+    tc = tcase_create("TC");
+    tcase_set_tags(tc, actual_str);
+    ret_val = tcase_get_tags(tc, ret_str, sizeof(ret_str));
+    
+    ck_assert_msg(ret_val >=  0,
+		  "Got illegal negative return value");
+
+    ck_assert_msg((unsigned int)ret_val == strlen(actual_str),
+	      "Got %d instead of %d for returned tag length",
+	      ret_val, strlen(actual_str));
+    
+    expected_str = strdup(actual_str);
+    expected_str[sizeof(ret_str) - 1] = '\0';
+
+    ck_assert_msg(strcmp(ret_str, expected_str) ==  0,
+	      "Got '%s' instead of '%s'",
+	      ret_str, expected_str);
+    free(expected_str);
+} END_TEST;
+
+START_TEST(set_get_zero_len)
+{
+    const char *expected_str = "Red Blue Green Yellow";
+    int ret_val;
+    TCase *tc;
+
+    tc = tcase_create("TC");
+    tcase_set_tags(tc, expected_str);
+    ret_val = tcase_get_tags(tc, NULL, 0);
+    
+    ck_assert_msg(ret_val >=  0,
+		  "Got illegal negative return value");
+
+    ck_assert_msg((unsigned int)ret_val == strlen(expected_str),
+	      "Got %d instead of %d for returned tag length",
+	      ret_val, strlen(expected_str));
+    
+} END_TEST;
+
+START_TEST(set_get_two_space)
+{
+    char ret_str[100];
+    const char *actual_str = "Red   Blue Green   Yellow";
+    const char *expected_str = "Red Blue Green Yellow";
+    int ret_val;
+    TCase *tc;
+
+    tc = tcase_create("TC");
+    tcase_set_tags(tc, actual_str);
+    ret_val = tcase_get_tags(tc, ret_str, sizeof(ret_str));
+    
+    ck_assert_msg(ret_val >=  0,
+		  "Got illegal negative return value");
+
+    ck_assert_msg((unsigned int)ret_val == strlen(expected_str),
+	      "Got %d instead of %d for returned tag length",
+	      ret_val, strlen(expected_str));
+    
+    ck_assert_msg(strcmp(ret_str, expected_str) ==  0,
+	      "Got '%s' instead of '%s'",
+	      ret_str, expected_str);
+} END_TEST;
+
+
+/*
+ * Show that with no filter we run all the tests
+ */
+START_TEST(null_filter)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, NULL, NULL, CK_VERBOSE);
+
+    ck_assert(test_executed[RED]);
+    ck_assert(test_executed[BLUE]);
+    ck_assert(test_executed[PURPLE]);
+    ck_assert(test_executed[YELLOW]);
+    ck_assert(test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(inc_nothing)
+{
+    reset_executed();
+
+    srunner_run(sr, NULL, NULL, "", NULL, CK_VERBOSE);
+
+    ck_assert(!test_executed[RED]);
+    ck_assert(!test_executed[BLUE]);
+    ck_assert(!test_executed[PURPLE]);
+    ck_assert(!test_executed[YELLOW]);
+    ck_assert(!test_executed[BLACK]);
+
+    reset_executed();
+
+} END_TEST
+
+START_TEST(exc_nothing)
```

Comment:
can you spell out the abbreviations for inc and exc? Initially when I looked at "exc" I misread it as "exe" and thought "the test executes nothing" which was incorrect.